app/models/minion.rb

```diff
@@ -18,6 +18,8 @@ def assign_role(role:)
       salt.assign_role role: role
     end
     true
+  rescue
```

Comment:
Actually this was just one of them. With HTTP lots of things can go wrong and we should be able to rescue from a lot more of possible exceptions (https://github.com/rainforestapp/http-exceptions/blob/master/lib/http/exceptions.rb#L7-L21). Should we rescue from all of them? A generic rescue could be hiding future problems and maybe it's not optimal.